tests/framework/randomness/test_index_map.py

```diff
@@ -149,58 +160,65 @@ def test_hash_uniformity(map_size_and_hashed_values):
 
 
 @pytest.fixture(scope="function")
-def index_map(mocker):
+def index_map(mocker: pytest_mock.MockFixture) -> type[IndexMap]:
     mock_index_map = IndexMap
 
-    def hash_mock(k, salt):
+    def hash_mock(
+        k: pd.Index[int], salt: pd.Series[datetime] | pd.Series[int] | pd.Series[float]
+    ) -> pd.Series[int]:
         seed = 123456
-        salt = IndexMap()._convert_to_ten_digit_int(pd.Series(salt, index=k))
-        rs = np.random.RandomState(seed=seed + salt)
+        digit_series = IndexMap()._convert_to_ten_digit_int(pd.Series(salt, index=k))
+        rs = np.random.RandomState(seed=seed + digit_series)
         return pd.Series(rs.randint(0, len(k) * 10, size=len(k)), index=k)
 
     mocker.patch.object(mock_index_map, "_hash", side_effect=hash_mock)
 
     return mock_index_map
 
 
-def test_update_empty_bad_keys(index_map):
+def test_update_empty_bad_keys(index_map: type[IndexMap]) -> None:
     keys = pd.DataFrame({"A": ["a"] * 10}, index=range(10))
     m = index_map(key_columns=list(keys.columns))
     with pytest.raises(RandomnessError):
         m.update(keys, pd.to_datetime("2023-01-01"))
 
 
-def test_update_nonempty_bad_keys(index_map):
+def test_update_nonempty_bad_keys(index_map: type[IndexMap]) -> None:
     keys = generate_keys(1000)
     m = index_map(key_columns=list(keys.columns))
     m.update(keys, pd.to_datetime("2023-01-01"))
     with pytest.raises(RandomnessError):
         m.update(keys, pd.to_datetime("2023-01-01"))
 
 
-def test_update_empty_good_keys(index_map):
+def test_update_empty_good_keys(index_map: type[IndexMap]) -> None:
     keys = generate_keys(1000)
     m = index_map(key_columns=list(keys.columns))
     m.update(keys, pd.to_datetime("2023-01-01"))
     key_index = keys.set_index(list(keys.columns)).index
-    assert len(m._map) == len(keys), "All keys not in mapping"
+    map = m._map
+    assert isinstance(map, pd.Series)
```

Comment:
I wonder if you need this assertion if you typed map specifically, `map: pd.Series = m._map`

Reply:
I specifically had to do this because _map can either be pd.Series[int] or None but in this case it has to be a series. For some reason mypy doesnt like inline typing vs the assertion.